core/src/main/scala/cats/data/Ior.scala

```diff
@@ -708,6 +708,17 @@ sealed abstract class Ior[+A, +B] extends Product with Serializable {
       (a, b) => that.fold(a2 => false, b2 => false, (a2, b2) => AA.eqv(a, a2) && BB.eqv(b, b2))
     )
 
+  final def compare[AA >: A, BB >: B](that: AA Ior BB)(implicit AA: Order[AA], BB: Order[BB]): Int =
+    (this, that) match {
+      case (Ior.Left(a1), Ior.Left(a2))         => AA.compare(a1, a2)
+      case (Ior.Left(_), _)                     => -1
+      case (Ior.Both(a1, b1), Ior.Both(a2, b2)) => Order[(AA, BB)].compare((a1, b1), (a2, b2))
```

Comment:
This is kind of nitpicky but it would be pretty easy to avoid the allocation of a new `Order` instance here.

Reply:
Do you think it's worth it for the loss of clarity? (Genuine question, not passive-aggressive 😂 )

Reply:
Looking at it again, it's actually at least three allocations (for the instance and then the tuples). Given that one of the places this could be used is for sorting collections, I'd be inclined to say three fewer object allocations per comparison is worth it.

Reply:
(I wouldn't block the PR for this, though.)

Reply:
👍